exonum/src/blockchain/transaction.rs

```diff
@@ -37,26 +37,33 @@ const TRANSACTION_STATUS_OK: u16 = MAX_ERROR_CODE + 1;
 // `Err(TransactionErrorType::Panic)`.
 const TRANSACTION_STATUS_PANIC: u16 = TRANSACTION_STATUS_OK + 1;
 
-/// Return value of the `Transaction`'s `execute' method. Changes made by the transaction are
-/// discarded if `Err` is returned, see `Transaction` documentation for the details.
+/// Returns the outcome of the `Transaction` `execute` method.
+///
+/// This outcome may be
+/// either the result of transaction execution or an `ExecutionError`, if execution has
+/// failed. Errors consist of an error code and an optional description.
 pub type ExecutionResult = Result<(), ExecutionError>;
 /// Extended version of `ExecutionResult` (with additional values set exclusively by Exonum
-/// framework) that can be obtained through `Schema`'s `transaction_statuses` method.
+/// framework) that can be obtained through `Schema` `transaction_statuses` method.
 pub type TransactionResult = Result<(), TransactionError>;
 
-/// Transaction processing functionality for `Message`s allowing to apply authenticated, atomic,
+/// Transaction processing functionality for `Message`, which allows applying authenticated, atomic,
 /// constraint-preserving groups of changes to the blockchain storage.
 ///
+/// A transaction in Exonum, is a group of sequential operations with the data.
+/// Transaction processing rules are defined in services; these rules determine
+/// the business logic of any Exonum-powered blockchain.
+///
 /// See also [the documentation page on transactions][doc:transactions].
 ///
 /// [doc:transactions]: https://exonum.com/doc/architecture/transactions/
 pub trait Transaction: Message + ExonumJson + 'static {
     /// Verifies the internal consistency of the transaction. `verify` should usually include
     /// checking the message signature (via [`verify_signature`]) and, possibly,
     /// other internal constraints. `verify` has no access to the blockchain state;
-    /// checks involving the blockchains state must be preformed in [`execute`](#tymethod.execute).
+    /// checks involving the blockchain state must be preformed in [`execute`](#tymethod.execute).
     ///
-    /// If a transaction fails `verify`,  is considered incorrect and cannot be included into
+    /// If a transaction fails, `verify` is considered incorrect and cannot be included into
```

Comment:
I guess author meant "If a transaction fails `verify`, it is considered incorrect". `verify` can't be considered incorrect, it's a function.